crates/bevy_ecs/src/archetype.rs

```diff
@@ -786,12 +786,7 @@ pub struct Archetypes {
 }
 
 /// Metadata about how a component is stored in an [`Archetype`].
-pub struct ArchetypeRecord {
-    /// Index of the component in the archetype's [`Table`](crate::storage::Table),
-    /// or None if the component is a sparse set component.
-    #[allow(dead_code)]
-    pub(crate) column: Option<usize>,
-}
+pub struct ArchetypeRecord {}
```

Comment:
This is a really messy state. I'd rather keep the dead code around, with an expect that links to an issue for what should be done to actually use it.